crates/auxmos/src/turfs/superconduct.rs

```diff
@@ -0,0 +1,490 @@
+			std::file!(),
+			std::line!(),
+			std::column!()
+		)
+	})? / 10.0) as f64;
+	let _ = sender.try_send(SSheatInfo { time_delta });
+	Ok(Value::null())
+}
+
+fn get_share_energy(delta: f32, cap_1: f32, cap_2: f32) -> f32 {
+	delta * ((cap_1 * cap_2) / (cap_1 + cap_2))
+}
+
+//Fires the task into the thread pool, once
+#[init(full)]
+fn _process_heat_start() -> Result<(), String> {
+	INIT_HEAT.call_once(|| {
+		rayon::spawn(|| loop {
+			//this will block until process_turf_heat is called
+			let info = with_heat_processing_callback_receiver(|receiver| receiver.recv().unwrap());
+			let task_lock = TASKS.read();
+			let start_time = Instant::now();
+			let sender = byond_callback_sender();
+			let emissivity_constant: f64 = STEFAN_BOLTZMANN_CONSTANT * info.time_delta;
+			let radiation_from_space_tick: f64 = RADIATION_FROM_SPACE * info.time_delta;
+			with_turf_heat_read(|arena| {
+				with_turf_gases_read(|air_arena| {
+					let adjacencies_to_consider = arena
+						.map
+						.par_iter()
+						.filter_map(|(&turf_id, &heat_index)| {
+							/*
+								If it has no thermal conductivity, low thermal capacity or has no adjacencies,
+								then it's not gonna interact, or at least shouldn't.
+							*/
+							let info = arena.get(heat_index).unwrap();
+							let temp = { *info.temperature.read() };
+							//can share w/ adjacents?
+							if arena.adjacent_heats(heat_index).any(|item| {
+								(temp - *item.temperature.read()).abs()
+									> MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER
+							}) {
+								return Some((turf_id, heat_index, true));
+							}
+							if temp > MINIMUM_TEMPERATURE_FOR_SUPERCONDUCTION {
+								//can share w/ space/air?
+								if info.adjacent_to_space
+									|| air_arena
+										.get_id(&turf_id)
+										.and_then(|&nodeid| {
+											air_arena.get(nodeid)?.enabled().then(|| ())
+										})
+										.is_some()
+								{
+									Some((turf_id, heat_index, false))
+								} else {
+									None
+								}
+							} else if let Some(node) = air_arena.get_id(&turf_id) {
+								let cur_mix = air_arena.get(*node).unwrap();
+								if !cur_mix.enabled() {
+									return None;
+								}
+								GasArena::with_all_mixtures(|all_mixtures| {
+									let air_temp = all_mixtures[cur_mix.mix].try_read();
+									if air_temp.is_none() {
+										return false;
+									}
+									let air_temp = air_temp.unwrap().get_temperature();
+
+									if air_temp < MINIMUM_TEMPERATURE_FOR_SUPERCONDUCTION {
+										return false;
+									}
+									(temp - air_temp).abs() > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER
+								})
+								.then(|| (turf_id, heat_index, false))
+							} else {
+								None
+							}
+						})
+						.filter_map(|(id, node_index, has_adjacents)| {
+							let info = arena.get(node_index).unwrap();
+							let mut temp_write = info.temperature.try_write()?;
+
+							//share w/ space
+							if info.adjacent_to_space && *temp_write > T0C {
```

Comment:
the boltzmann method actually cools too slow, now. because turfs only process heat when it's above a certain temp